var/spack/repos/builtin/packages/fenics-basix/package.py

```diff
@@ -23,3 +23,6 @@ class FenicsBasix(CMakePackage):
     depends_on("xtl@0.7.2:")
     depends_on("xtensor@0.23.10:")
     depends_on("blas", type=("build", "run"))
+
+    with when("@main"):
+        root_cmakelists_dir = "cpp"
```

Comment:
The `when` context manager works only with directives. Does this have any effect for you?

Reply:
@alalazo - I am not sure. It seems to work OK as it is. I was not sure how else to set a variable, which depends on the version.

Reply:
I guess it works also without the `when`. Did you try both `main` and other versions?

> I was not sure how else to set a variable, which depends on the version.

In Python you can code a "property":
```python
@property
def root_cmake_lists_dir(self):
    if self.spec.satisfies("@main"):
        return "cpp"
    return None
```

Reply:
Thanks - I have updated to use this.

Reply:
@alalazo is anything more needed for the PR?